dmutils/status.py

```diff
@@ -47,3 +48,67 @@ def enabled_since(date_string):
         return date_string
 
     return False
+
+
+class StatusError(Exception):
+    """A stub class to use when implementing additional checks for an app's _status endpoint. See API for example."""
+    message = ''
+
+    def __init__(self, message):
+        super().__init__(message)
+        self.message = message
+
+
+def get_app_status(data_api_client=None,
+                   search_api_client=None,
+                   ignore_dependencies=False,
+                   additional_checks=None):
+    """Generates output for `_status` endpoints on apps
+
+    :param current_app: The flask `current_app` global.
+    :param data_api_client: The app's data_api_client, if used.
+    :param search_api_client: The app's search_api_client, if used.
+    :param ignore_dependencies: Minimal endpoint checks only (i.e. the app is routable and disk space is fine).
+    :param additional_checks: A sequence of callables that return dicts of data to be injected into the final JSON
+                              response or raise StatusErrors if they need to log an error that should fail the
+                              check (this will cause the endpoint to return a 500).
+    :return: A dictionary describing the current state of the app with, at least, a 'status' key with a value of 'ok'
+             or 'error'.
+    """
+    error_messages = []
+    core_data = {'status': 'ok'}  # Core checks (endpoint resolves, disk is OK)
+    extra_data = {}  # Data from additional checks to run (if ignore_dependencies is False), e.g. api connectivity
+
+    disk_status = get_disk_space_status()
+    core_data['disk'] = f'{disk_status[0]} ({disk_status[1]}% free)'
+    if disk_status[0].lower() != 'ok':
+        error_messages.append(f'Disk space low: {disk_status[1]}% remaining.')
+
+    if not ignore_dependencies:
+        extra_data['version'] = current_app.config['VERSION']
+        extra_data['flags'] = get_flags(current_app)
+
+        if data_api_client:
+            data_api_status = data_api_client.get_status() or {}
+            extra_data['api_status'] = data_api_status
+            if data_api_status.get('status').lower() != 'ok':
```

Comment:
Get will return `None` if the thing isn't there, and `.lower` on `None` won't work, so there's an untested edge-case here and the logic isn't quite right.

Reply:
Yeah I've just been fixing this up. You snuck in over lunch 🙈

Reply:
The weirdness is that api client's `get_status` can return None in some edge case (where the apiclient is disabled, all requests return None). The API was testing this specifically so I've come back here to fix it up and you caught me in the middle of that process. I've updated it now to provide a default dict if `get_status` does happen to return None.